src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -40,9 +42,16 @@ private enum StreamState : byte
             private int _pendingWindowUpdate;
 
             private StreamState _state;
-            private TaskCompletionSource<bool> _waiterTaskSource;
             private bool _disposed;
 
+            /// <summary>The core logic for the IValueTaskSource implementation.</summary>
+            private ManualResetValueTaskSourceCore<bool> _waitSourceCore = new ManualResetValueTaskSourceCore<bool> { RunContinuationsAsynchronously = true }; // mutable struct, do not make this readonly
```

Comment:
Naming: maybe just "_waitSource"? I realize the class has Core in its name, but I'm not sure what having that on the field name adds.

Reply:
Ok